src/MobileUI/ViewModels/EarnDetailsViewModel.cs

```diff
@@ -135,12 +136,17 @@ await _quizService.SubmitAnswer(new SubmitQuizAnswerDto()
             }
             
             MoveNext(Questions.IndexOf(CurrentQuestion) + 1);
+            IsBusy = false;
         }
 
         private async Task SubmitResponses()
         {
-            await SubmitAnswer();
             bool allQuestionsAnswered = Questions.All(q => q.IsSubmitted);
+
+            if (IsLastQuestion && !allQuestionsAnswered)
+            {
+                await SubmitAnswer();
+            }
             
             if (allQuestionsAnswered)
```

Comment:
Here `allQuestionAnswered` will always be `false` because you didn't re-evaluate the value after submitting the last answer.

Reply:
Thanks, I've fixed this up